### PR TITLE
Fix completion issues in language server

### LINKED ISSUE #28198: ballerinai/transaction should be restricted from importing
**Description:**
transaction_ai

**Suggested Labels:**

**Suggested Assignees:**

**Affected Product Version:**

**OS, DB, other environment details and versions:**    

**Steps to reproduce:**

**Related Issues:**

### LINKED ISSUE #29549: `Couldn't find the invocation symbol!` in output window with `if` having empty brackets
**Description:**
$title
ezgif com-gif-maker (6)

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by nadeeshaan**
## Purpose
Fixes #28198
Fixes #29549

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by IMS94 — APPROVED** [review]
LGTM!

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ConstantDeclarationNodeContext.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://wso2.com) All Rights Reserved.
```
Comment (IMS94) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ConstantDeclarationNodeContext.java:2]: This should be 2021 I guess.

File: language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/completion/ModuleConstantContextTest.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (IMS94) [inline language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/completion/ModuleConstantContextTest.java:2]: Same. Should be 2021?

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ConstantDeclarationNodeContext.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    private boolean onTypeDescContext(BallerinaCompletionContext context, ConstantDeclarationNode node) {
+        int cursor = context.getCursorPositionInTree();
+        Token constKeyword = node.constKeyword();
+        Token variableName = node.variableName();
+        Optional<TypeDescriptorNode> tDesc = node.typeDescriptor();
+
+        if (cursor < constKeyword.textRange().endOffset() + 1) {
+            return false;
+        }
+        if (tDesc.isPresent()) {
```
Comment (IMS94) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ConstantDeclarationNodeContext.java:97]: Not sure if it's personal preference, what if we use `if-elseif` instead?

**Review by IMS94 — APPROVED** [review]
Approved with minor comments.

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ConstantDeclarationNodeContext.java (hunk 3 above)
Comment (dulajdilshan) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ConstantDeclarationNodeContext.java:97]: This is the early-return approach for coding. My idea is this is more readable than having `if-elseif`'s 
https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement

Comment (IMS94) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ConstantDeclarationNodeContext.java:97]: @dulajdilshan Agree on returning early. I was just saying to use:
```java
        if (cursor < constKeyword.textRange().endOffset() + 1) {
            return false;
        } else if (tDesc.isPresent()) {
            return true;
        }...
```

**Review by mohanvive — APPROVED** [review]

**Pull request merged**